android/src/main/kotlin/com/amplitude/amplitude_flutter/AmplitudeFlutterPlugin.kt

```diff
@@ -14,10 +15,14 @@ import io.flutter.plugin.common.MethodCall
 import io.flutter.plugin.common.MethodChannel
 import io.flutter.plugin.common.MethodChannel.MethodCallHandler
 import io.flutter.plugin.common.MethodChannel.Result
+import android.content.pm.PackageManager
+import com.amplitude.android.utilities.DefaultEventUtils
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
 
-class AmplitudeFlutterPlugin : FlutterPlugin, MethodCallHandler {
+class AmplitudeFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
     lateinit var amplitude: Amplitude
-
+    private var activity: Activity? = null
```

Comment:
Does Flutter have weak references? Might want to use one here to prevent memory leaks.

Reply:
Activities in Android are heavyweight objects can potentially cause memory leaks. Use [WeakReference](https://developer.android.com/reference/java/lang/ref/WeakReference) to point to activity. Update in https://github.com/amplitude/Amplitude-Flutter/pull/173/commits/5a042bec05fdf64c18d04a948b0bdecad661da86